components/brave_ads/browser/ads_service_impl.cc

```diff
@@ -458,6 +458,10 @@ void AdsServiceImpl::MaybeStartBatAdsService() {
     return;
   }
 
+  if (bat_ads_service_.is_bound()) {
```

Comment:
nit: should we not check this before line 451?

Reply:
yeah, we can do this check earlier